### PR TITLE
Enter structured text

### PR DESCRIPTION
**Pull request by laclark22**
For JIRA 1770. [Pilot 1] Typing text in the middle of a structured field and then hitting enter does not properly move the cursor

_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

When you enter text inside of structured text and click enter the cursor now moves to the next line. 

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [x] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [x] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [x] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [ ] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by Dtphelan1** [comment]
I think this looks good to me! Functionality works as specified and didn't identify anything else weird about the special case we've added.

File: src/notes/StructuredFieldPlugin.jsx [hunk 1]
```diff
@@ -86,6 +86,23 @@ function StructuredFieldPlugin(opts) {
             }
         }
 
+        if(!(shortcut) && e.key === 'Enter') {
```
Comment (jafeltra) [inline src/notes/StructuredFieldPlugin.jsx]: There should be a space after the `if` here.

File: src/notes/StructuredFieldPlugin.jsx [hunk 2]
```diff
@@ -86,6 +86,23 @@ function StructuredFieldPlugin(opts) {
...
+
+            const beforeAnchorKey = state.document.getPreviousSibling(selection.anchorKey);
+            const afterAnchorKey = state.document.getNextSibling(selection.anchorKey);
+            const currentNode = state.document.getNode(selection.anchorKey);
+            
+            if ((beforeAnchorKey && afterAnchorKey) && beforeAnchorKey.type === "structured_field" && afterAnchorKey.type === "structured_field" && currentNode.text.length === selection.anchorOffset){
```
Comment (jafeltra) [inline src/notes/StructuredFieldPlugin.jsx]: There should be a space before the `{`. Also the blank line above has extra spaces it doesn't need.

**Comment by jafeltra** [comment]
Just pushed the changes for JA1. That should be set!

**Comment by gregquinn2001** [comment]
JA2 definitely sounds in scope for JIRA 1770 so should be addressed as part of this pull request.

**Comment by gregquinn2001** [comment]
Nice find Julia!

**Comment by Dtphelan1** [comment]
Good catch on the styling stuff and the unique enter case Jules!

**Comment by laclark22** [comment]
[JA2] has been addressed.

**Pull request merged**